src/pyfmi/fmi_algorithm_drivers.py

```diff
@@ -927,6 +932,8 @@ def __init__(self,
         self.input = input
 
         self.status = 0
+        # used for keeping track of when to write data when downsampling the simulation results
+        self.downsampling_counter = 0
```

Comment:
I would call it something different, since this counter represents the # of the step we are in.

On second thought, I think it may even be easier to iterate over the grid (`for t in grid:`) using `for i, t in enumerate(grid):` instead, then `i` (maybe with better name) becomes what is now `self.downsampling_counter`.

Could even skip the extra function and directly put it into the loop in `solve`.